internal/publish/model/stats.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package model is a model abstraction for health authority telemetry.
+package model
+
+import (
+	"time"
+)
+
+const (
+	MaxOldestTEK = 14
+	MaxOnsetDays = 28
+
+	PlatformAndroid = "android"
+	PlatformIOS     = "ios"
+	PlatformUnknown = "unknown"
+)
+
+var (
+	platforms = map[string]int{
+		PlatformAndroid: 0,
+		PlatformIOS:     1,
+		PlatformUnknown: 2,
+	}
+)
+
+// HealthAuthorityStats represents the raw metrics for an individual
+// health authority for a given hour.
+type HealthAuthorityStats struct {
+	HealthAuthorityID int64
+	Hour              time.Time
+	PublishCount      []int32
```

Comment:
Using int32 here feels like a premature optimization that's going to come back and bite us. I made some premature optimizations with the modeler and I would prefer if we didn't make the same mistake again.

Reply:
I widened these to int64/ bigint